includes/Blocks/Block.php

```diff
@@ -237,72 +357,93 @@ private function resolve( $block ) {
+					} else {
+						$result[ $key ] = DOMHelpers::parseHTML( $html, $value['selector'] );
+	
+						if ( isset( $value['multiline'] ) && ! empty( $result[ $key ] ) ) {
+							$result[ $key ] = DOMHelpers::getElementsFromHTML( $result[ $key ], $value['multiline'] );
+						}
+					}
 					break;
-			}
-			return $value;
-		}
-		// Case when both selector and source are defined
-		if ( isset( $attribute_config['selector'], $attribute_config['source'] ) ) {
-			$rendered_block = wp_unslash( render_block( $block ) );
-			$value          = null;
-			if ( empty( $rendered_block ) ) {
-				return $value;
-			}
-
-			switch ( $attribute_config['source'] ) {
 				case 'attribute':
-					$value = DOMHelpers::parseAttribute( $rendered_block, $attribute_config['selector'], $attribute_config['attribute'], $default );
-					break;
-				case 'html':
-					$value = DOMHelpers::parseHTML( $rendered_block, $attribute_config['selector'], $default );
-
-					if ( isset( $attribute_config['multiline'] ) && ! empty( $value ) ) {
-						$value = DOMHelpers::getElementsFromHTML( $value, $attribute_config['multiline'] );
-					}
-
+					$result[ $key ] = DOMHelpers::parseAttribute( $html, $value['selector'], $value['attribute'], $value );
 					break;
 				case 'text':
-					$value = DOMHelpers::getTextFromSelector( $rendered_block, $attribute_config['selector'], $default );
-
+					$result[ $key ] = DOMHelpers::parseText( $html, $value['selector'] );
 					break;
-			}//end switch
+				case 'query':
+					$temp = [];
+					foreach ( DOMHelpers::findNodes( $html, $value['selector'] ) as $source_node ) {
+						foreach ( $value['query'] as $q_key => $q_value ) {
+							$temp_config    = [
+								$q_key => $q_value,
+							];
+							$res            = $this->resolve_block_attributes_recursive( $attributes, $source_node->html(), $temp_config );
+							$temp[ $q_key ] = $res[ $q_key ];
+						}
+						$result[ $key ][] = $temp;
+					}
+					break;
+			}
 			// Post processing of return value based on configured type
-			switch ( $attribute_config['type'] ) {
+			switch ( $value['type'] ) {
 				case 'integer':
-					$value = intval( $value );
+					$result[ $key ] = intval( $result[ $key ] );
 					break;
 				case 'boolean':
-					// If the value is empty or false return
-					if ( is_null( $value ) || false === $value ) {
+					if ( false === $result[ $key ] ) {
 						break;
 					}
-					// Otherwise it's truthy
-					$value = true;
+					if ( is_null( $result[ $key ] ) ) {
+						$result[ $key ] = false;
+						break;
+					}
+					$result[ $key ] = true;
 					break;
 			}
+			
+			if ( empty( $result[ $key ] ) ) {
+				$result[ $key ] = $attributes[ $key ] ?? $default;
+			}
+		}
+		
+		return $result;
+	}
 
+	/**
+	 * Parses the block content of a source only block type
+	 * 
+	 * @param string $html The html value
+	 * @param string $source The source type
+	 */
+	private function parse_single_source( $html, $source ): string|null {
```

Comment:
Union types like `string|null` are not compat with PHP 7.4.

Reply:
typing is hard :)

Reply:
@mindctrl @theodesp both PHPCompatibilityWP and phpstan should have caught this. I think there's an issue with the linter/ci 🤔